xtras/src/supervisor.rs

```diff
@@ -280,6 +301,48 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn supervisor_can_delay_respawn() {
```

Comment:
:+1: Very clear test! Thanks.

The second assertion makes me a bit nervous because it relies on relatively tricky timing, but I see no alternative!

Reply:
I've tried a few alternatives, but could not find anything easier :/